File: 0714. Best Time to Buy and Sell Stock with Transaction Fee/Solution.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices, int fee)
    {
        int f0 = INT_MIN, f1 = 0;
        for (int price : prices) {
            f0 = max(f1 - price, f0);
            f1 = max(f1, f0 + price - fee);
        }
        return f1;
    }
};

File: 0714. Best Time to Buy and Sell Stock with Transaction Fee/Solution.py
from math import inf
from typing import List


class Solution:
    def maxProfit(self, prices: List[int], fee: int) -> int:
        f0, f1 = -inf, 0
        for price in prices:
            f0 = max(f0, f1 - price)
            f1 = max(f1, f0 + price - fee)
        return f1
